src/validation.cpp

```diff
@@ -1837,10 +1837,11 @@ class WarningBitsConditionChecker : public AbstractThresholdConditionChecker
 public:
     explicit WarningBitsConditionChecker(int bitIn) : bit(bitIn) {}
 
-    int64_t BeginTime(const Consensus::Params& params) const override { return 0; }
-    int64_t EndTime(const Consensus::Params& params) const override { return std::numeric_limits<int64_t>::max(); }
+    int StartHeight(const Consensus::Params& params) const override { return 0; }
+    int TimeoutHeight(const Consensus::Params& params) const override { return std::numeric_limits<int>::max(); }
     int Period(const Consensus::Params& params) const override { return params.nMinerConfirmationWindow; }
-    int Threshold(const Consensus::Params& params) const override { return params.nRuleChangeActivationThreshold; }
+    int Threshold(const Consensus::Params& params) const override { return params.m_vbits_min_threshold; }
```

Comment:
The warning message emitted when this threshold is reached is "Warning: unknown new rules activated", whereas now the threshold being reached really means "Warning: unknown new rules may be activated soon" (since 75% doesn't actually indicate that any vbits deployment will activate).

I think it might make sense to remove the `m_vbits_min_threshold` parameter from chain params and set the threshold to 75% of nMinerConfirmationWindow here.

Reply:
Perhaps for a followup, or if I retouch.